#! /bin/sh /usr/share/dpatch/dpatch-run
## 16fix_gcc_warnings.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Get rid of several minor gcc warnings.

@DPATCH@
diff -Naur screen-4.0.3.orig/attacher.c screen-4.0.3/attacher.c
--- screen-4.0.3.orig/attacher.c	2008-06-01 23:40:23.000000000 +0200
+++ screen-4.0.3/attacher.c	2008-06-01 23:41:01.000000000 +0200
@@ -820,7 +820,7 @@
   pam_handle_t *pamh = 0;
   int pam_error;
 #endif
-  char *pass, mypass[16 + 1], salt[3];
+  char *pass = 0, mypass[16 + 1], salt[3];
   int using_pam = 1;
 
 #ifdef USE_PAM
diff -Naur screen-4.0.3.orig/layer.c screen-4.0.3/layer.c
--- screen-4.0.3.orig/layer.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/layer.c	2008-06-01 23:41:01.000000000 +0200
@@ -451,7 +451,7 @@
 void
 LClearLine(l, y, xs, xe, bce, ol)
 struct layer *l;
-int xs, xe, bce;
+int y, xs, xe, bce;
 struct mline *ol;
 {
   struct canvas *cv;
diff -Naur screen-4.0.3.orig/process.c screen-4.0.3/process.c
--- screen-4.0.3.orig/process.c	2008-06-01 23:40:40.000000000 +0200
+++ screen-4.0.3/process.c	2008-06-01 23:41:01.000000000 +0200
@@ -3123,7 +3123,7 @@
           int newnr, fl = 0, kf = 0, af = 0, df = 0, mf = 0;
 	  struct display *odisp = display;
 	  int used = 0;
-          struct kmap_ext *kme;
+          struct kmap_ext *kme = 0;
 
 	  for (; *args && **args == '-'; args++, argl++)
 	    {
diff -Naur screen-4.0.3.orig/screen.c screen-4.0.3/screen.c
--- screen-4.0.3.orig/screen.c	2008-06-01 23:39:49.000000000 +0200
+++ screen-4.0.3/screen.c	2008-06-01 23:41:01.000000000 +0200
@@ -2360,7 +2360,7 @@
   int truncpos = -1;
   int truncper = 0;
   int trunclong = 0;
-  struct backtick *bt;
+  struct backtick *bt = 0;
  
   if (winmsg_numrend >= 0)
     winmsg_numrend = 0;
